src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -236,12 +242,12 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
                 return document;
             }
 
-            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+            var fixAllService = document.Project.Solution.Workspace.Services.GetRequiredService<IFixAllGetFixesService>();
 
             var solution = await fixAllService.GetFixAllChangedSolutionAsync(
                 fixCollection.FixAllState.CreateFixAllContext(progressTracker, cancellationToken)).ConfigureAwait(false);
 
-            return solution.GetDocument(document.Id);
+            return solution.GetDocument(document.Id) ?? document;
```

Comment:
💭 I find the new code misleading. Is it possible for the fallback code to ever execute?

Reply:
What if the code fix removes the document from the solution? Just being defensive here...

Reply:
Is this the Fix All in Document path? Sounds like this case would produce a silent failure of the code fix right?

Reply:
The API only seems to be used from code cleanup path. Would you prefer I assert non-null document in fixed solution or update the API and call site to handle null? Not sure how exactly the call site should be update though if one of the fixes removes the document...

Reply:
if it is expected that the doc could be remove,d tehn we have to handle that (and not fallback to the old doc right)?  If it's not expected, then we should throw, right?  Silently replacing null with the old doc doesn't seem correct in either situation.

Reply:
I have changed this to throw a NotSupportedException if document gets removed. We can tweak this later if Code cleanup supports fixes that can remove a document.